tests/packagedcode/test_swift.py

```diff
@@ -101,3 +101,24 @@ def test_package_scan_swift_end_to_end_full_mapboxmaps_resolved_only(self):
         check_json_scan(
             expected_file, result_file, remove_uuid=True, regen=REGEN_TEST_FIXTURES
         )
+
+    def test_package_scan_swift_end_to_end_full_mapboxmaps_resolved_only(self):
+        test_dir = self.get_test_loc("packages/vercelui")
```

Comment:
This is a nit. Could be possibly improved later too.

1. Let's not only add full-scan tests. This is much harder to test and review in general, and also takes up much more space. Instead do individual tests of `parse()` and `is_datafile()` functions (at least one test per parser function). One test should only check one thing that can go wrong. It's fine to have a couple assembly functions later, but using barebones manifests just enough for assembly.
2. Let's truncate test files (like big the ones with dependency trees) to reduce file size for the tests, we have a very large number of tests and this would keep increasing the git checkout size/source size more rapidly that we can manage. Let's either find smaller examples, or reduce the files so that there is just enough to test the features.

Reply:
@AyanSinhaMahapatra I've truncate the large test file. Will add more individual tests along with the fix for https://github.com/nexB/scancode-toolkit/issues/3837 in next PR.